src/reducers/versions.spec.tsx

```diff
@@ -3235,6 +3324,100 @@ describe(__filename, () => {
         currentVersionId: versionId,
       });
     });
+
+    it('syncs selectedPath in version object to the one out of version', () => {
+      const path1 = 'path/1';
+      const path2 = 'path/2';
+      const oldVersion = createExternalVersionWithEntries([{ path: path1 }], {
+        id: nextUniqueId(),
+        selected_file: path1,
+      });
+      const newVersion = createExternalVersionWithEntries(
+        [{ path: path1 }, { path: path2 }],
+        {
+          id: nextUniqueId(),
+          selected_file: path2,
+        },
+      );
+
+      let state = reducer(
+        undefined,
+        actions.loadVersionInfo({ version: newVersion }),
+      );
+      state = reducer(state, actions.loadVersionInfo({ version: oldVersion }));
+      state = reducer(
+        state,
+        actions.setCurrentVersionId({ versionId: newVersion.id }),
+      );
+
+      expect(state).toHaveProperty('selectedPath', path1);
+      expect(state).toHaveProperty(
+        `versionInfo.${newVersion.id}.selectedPath`,
+        state.selectedPath,
+      );
+    });
+
+    it('sets selectedPath to version.selected if it  does not exist in new version', () => {
```

Comment:
This property name should be explicit just to avoid confusion:

```suggestion
    it('sets selectedPath to version.selectedPath if it does not exist in the new version', () => {
```